drivers/scheduler/oke/oke.go

```diff
@@ -0,0 +1,585 @@
+	Data []struct {
+		DisplayName  string `json:"display-name"`
+		Id           string `json:"id"`
+		FreeformTags struct {
+			K8SVersion string `json:"k8s_version"`
+		} `json:"freeform-tags,omitempty"`
+	} `json:"data"`
+}
+
+func (o *oke) String() string {
+	return SchedName
+}
+
+// Init initializes the node driver for oke under the given scheduler
+func (o *oke) Init(schedOpts scheduler.InitOptions) error {
+
+	instanceGroup := os.Getenv("INSTANCE_GROUP")
+	if len(instanceGroup) != 0 {
+		o.instanceGroupName = instanceGroup
+	} else {
+		o.instanceGroupName = "default"
+	}
+	compartmentId := os.Getenv("PX_ORACLE_compartment_id")
+	if len(compartmentId) == 0 {
+		return fmt.Errorf("compartment id not provided as env var [PX_ORACLE_compartment_id]: %s", compartmentId)
+	}
+	o.compartmentId = compartmentId
+
+	ops, err := oracleOps.NewClient()
+	if err != nil {
+		return err
+	}
+	o.ops = ops
+	err = o.K8s.Init(schedOpts)
+	if err != nil {
+		return err
+	}
+
+	err = configureUser()
+	if err != nil {
+		return err
+
+	}
+	err = o.setCluster()
+
+	return err
+}
+
+func configureUser() error {
+	userKey := os.Getenv("PX_ORACLE_user_ocid")
+	if len(userKey) == 0 {
+		return fmt.Errorf("user key not provided as env var [PX_ORACLE_user_oci]: %s", userKey)
+	}
+
+	fingerprint := os.Getenv("PX_ORACLE_fingerprint")
+	if len(fingerprint) == 0 {
+		return fmt.Errorf("fingerprint key not provided as env var [PX_ORACLE_fingerprint]: %s", fingerprint)
+	}
+	privateKey := os.Getenv("PX_ORACLE_private_key_path")
+	if len(privateKey) == 0 {
+		return fmt.Errorf("private key not provided as env var [PX_ORACLE_private_key_path]: %s", privateKey)
+	}
+	region := os.Getenv("PX_ORACLE_cluster_region")
+	if len(region) == 0 {
+		return fmt.Errorf("region not provided as env var [PX_ORACLE_cluster_region]: %s", region)
+	}
+	tenancy := os.Getenv("PX_ORACLE_tenancy")
+	if len(tenancy) == 0 {
+		return fmt.Errorf("tenancy not provided as env var [PX_ORACLE_tenancy]: %s", tenancy)
+	}
+
+	creatDir := fmt.Sprintf("mkdir %s", "/root/.oci")
+	_, stderr, err := osutils.ExecShell(creatDir)
+	if err != nil {
+		return fmt.Errorf("error in creating .oci directory, error %v %v", err, stderr)
+	}
+
+	cmd := fmt.Sprintf("echo '[DEFAULT]\nuser=%s\nfingerprint=%s\ntenancy=%s\nregion=%s\nkey_file=%s' > %s", userKey, fingerprint, tenancy, region, privateKey, okeConfigFile)
+	_, stderr, err = osutils.ExecShell(cmd)
+	if err != nil {
+		return fmt.Errorf("error in configuring User info in OCI CLI, error %v %v", err, stderr)
+	}
+
+	cmd = fmt.Sprintf("sudo chmod 400 %s", okeConfigFile)
+	_, stderr, err = osutils.ExecShell(cmd)
+	if err != nil {
+		return fmt.Errorf("error in setting permissions for oci config, error %v %v", err, stderr)
+	}
+
+	return nil
+}
+
+func (o *oke) setCluster() error {
+	clusterName := os.Getenv("PX_ORACLE_cluster_name")
+	if len(clusterName) == 0 {
+		return fmt.Errorf("cluster name not provided as env var [PX_ORACLE_cluster_name]: %s", clusterName)
+	}
```

Comment:
No reason to print `clusterName` if we know its for sure empty string

Reply:
fixed